CONTRIBUTING.md

```diff
@@ -318,6 +318,14 @@ That means the following conditions must hold.
 
 See [the separate coding style document](CODINGSTYLE.md).
 
+Formatting of the code is verified automatically in CI with various tools.
+C++ code is checked with `clang-format`, which can also update files in-place.
+Use `tools/docker/clang-format.sh` to make sure formatting is correct before each commit.
+Python code is checked with `flake8`.
+Use the command `docker run -v $PWD:$PWD pipelinecomponents/flake8:f087c4c flake8 $PWD/tests/opts.py` from the top-level directory to check the file `tests/opts.py`.
+Do this for every file you modified.
```

Comment:
Perhaps we should note that auto-generated files aren't included, and they clang-format has an exclusion rule for them.